extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -209,6 +209,30 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
             //ignore, Apicurio Avro is not in the classpath
         }
 
+        try {
+            Class.forName("io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler", false,
+                    Thread.currentThread().getContextClassLoader());
+
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
+                    "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler",
+                    "org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerRefreshingLogin",
+                    "org.apache.kafka.common.security.oauthbearer.OAuthBearerToken"));
```

Comment:
Wondering if these 2 last classes shouldn't be always added. WDYT?

Reply:
> About `Subject`, we had a similar issue with Jass and Kafka. We need to check why the lookup fails. IT generally because of a missing class or module.

If you have any pointers on what to try and how to more info please let me know. I can try to provide substitutions for some of these classes and see if that would help.

Reply:
I moved them outside of the strimzi oauth block, so they can be used for general oauth use cases as well.